components/http/okHttp/src/main/java/com/microsoft/kiota/http/OkHttpRequestAdapter.java

```diff
@@ -614,6 +644,15 @@ private Response throwIfFailedResponse(
             final int statusCode = response.code();
             final ResponseHeaders responseHeaders =
                     HeadersCompatibility.getResponseHeaders(response.headers());
+            if (statusCode >= 300 && statusCode < 400) {
+
+                if (responseHeaders.get("Location") != null) {
+                    spanForAttributes.setAttribute("Location", true);
+                    return response;
+                }
+                spanForAttributes.setAttribute(errorMappingFoundAttributeName, false);
+                return null;
```

Comment:
Shoudn't we return the reponse either way to the caller since we aren't throwing an exception?